### ISSUE TITLE
l10n_uk

### ISSUE THREAD

**Issue by lap-odoo** [odoo/odoo#18510]
Dear Odoo Partners, 

At Odoo, a lot of effort is put into **improving the accounting localizations** for each country and also to **maintain these accounting localizations up-to-date**. To achieve that, we will need your help. 

One github issue will be created per accounting localization (e.g. l10n_be for Belgium). We expect from you to make suggestions about what should be improved in the specific accounting localization. Do not hesitate to share files with **detailed explanations**. 

As soon as changes will be implemented, you will be informed and will have the opportunity **to test it and share your feedback**. 

This github issue will also be useful to **keep you informed** about all the ongoing improvements concerning a specific accounting localization.

**Comment by lap-odoo** [comment]
Dear Odoo UK Partners, 

First, I would like to know if changes are needed on the l10n_uk module, more specifically on: 
- The CoA
- The Taxes
- The Fiscal Positions
- The UK Legal Statements

Thanks a lot for you help

**Comment by guanhuan** [comment]
Hi Laura,

since Odoo is improving its localisation for UK recently, I am wondering if Odoo deals with VAT late claims.

Please find the following links regarding how Xero deals with it:
https://help.xero.com/uk/Report_VATReturn$BK_LateClaims
https://help.xero.com/uk/VAT-LateClaims
https://help.xero.com/uk/Report_VATAudit

We can provide some detailed spec if needed.

Please let me know if you have any question or need any help. 

Best

**Comment by lap-odoo** [comment]
@guanhuan : Thanks for the suggestion. It's not in our pipe for now but we may included this in the future versions. If I need additional information I'll come back to you. 

But first, we would like to make sure that the basic requirements are correctly fulfilled : CoA, Taxes, Fiscal Positions, Legal Reports (P&L, tax, BS, Etc.). Do you think that what we currently have is complete ?

**Comment by avw-odoo** [comment]
@guanhuan we are finishing "Improve the UK tax return (compare with XERO)"
and "Correct taxes (compare with XERO)" in master. I'll keep you posted when you can do some tests.

Best,

**Comment by avw-odoo** [comment]
Dear Odoo UK Partners,

we are seeking help to describe the requirements in order to comply with Tax Digital for VAT legislation.

Thanks a lot for you help

**Comment by StuartJMackintosh** [comment]
Alain (@avw-odoo), we have set the UK finance up for many companies on Odoo and this includes the chart of accounts (BS & PL), tax, fiscal positions, EC sales list and intrastat. 

I will invite a member of our Odoo specialist team to join this thread and share some of the UK localisation work we have done and changes we have made to make UK implementation easier.

If you have any specific accounting questions or topics, let me know and I can put to our consultants team.

All the best,
Stuart.

**Comment by avw-odoo** [comment]
Hello @StuartJMackintosh, regarding the changes you have made to make UK implementation easier:

 - For which Odoo version is ?
 - Can you summarize the gap between your work and the Odoo enterprise standard l10n_uk ?
 - Do you think we should merge your improvements in Odoo standard ?
 - Do you have a github branch with your work ?

Best,

**Comment by StuartJMackintosh** [comment]
Hello @avw-odoo 

1. V8, V9, V10, V11 (All of the [OpusVL](http://opusvl.com/) work is based on the community editions.)
2. Stationary updates, configuration (fiscal position, CoA, tax, reports), additional reports for UK
3. You are welcome
4. Everything is within Git, we have modules that apply the changes. The developers would have to work out with you how to prepare a branch that would be easy to merge

We are constantly improving the modules and we have a well qualified accountant on the team who has a background of Sage and client input with Xero experience. The ultimate migration from Sage, Xero (and Netsuite, IFS etc) seems to work well for the client finance teams.

**Comment by StuartJMackintosh** [comment]
@ManishS-OpusVL - As discussed, see above.

**Comment by avw-odoo** [comment]
@StuartJMackintosh Thank you. With a pull request we can move forward to check how/what we can integrate from your improvements.

Do you have advice / feedback regarding Tax Digital for VAT legislation ?

**Comment by ManishS-OpusVL** [comment]
Dear Community Members,

Firstly apologies for the delay as I have been too tied up with other Odoo work.

So as mentioned by @StuartJMackintosh I would like to provide here an idea of the work we have been doing on Odoo UK Localisation.

Some of the key UK Localisation items we have done and implemented for our other customers are specified below:

**Comment by avw-odoo** [comment]
Hello @ManishS-OpusVL 

Do you think interesting to improve the l10n_uk standard module from your developments ?

**Comment by ManishS-OpusVL** [comment]
Hello @avw-odoo 

Yes, certainly. We have already scoped some significant improvements in the existing l10n_uk standard module.

One of the key one is related to the terminology used for Tax codes. We can come up with entire list when we are ready to publish our improvised UK standard module.

Further would like to mention that we have an experienced Chartered Accountant in our team who is working alongwith me to provide his expert financial insight to the project and to ensure that all the accounting related improvements are compliant as per the UK standard.

Thanks,
Manish Shukla

**Comment by cierek** [comment]
**MAKING TAX DIGITAL**

Current behavior:
Users can display and download tax documents in Excel/PDF
Expected behavior:
Automated/one click process from April 2019

The Making Tax Digital requirements will apply to all VAT registered businesses for their VAT obligations from April 2019 where their turnovers are in excess of the VAT threshold (currently £85,000).
From April 2019, VAT registered businesses (including self-employed and landlords) with turnover above the VAT registration threshold (currently £85,000) will have to:
•	keep their records digitally (for VAT purposes only)
•	provide their VAT return information to HMRC through Making Tax Digital compatible software

Overview:
https://www.gov.uk/government/publications/making-tax-digital/overview-of-making-tax-digital
Compatible software suppliers:
https://www.gov.uk/guidance/software-for-sending-income-tax-updates
Making tax digital centre:
https://www.aat.org.uk/making-tax-digital

**Comment by arveitch** [comment]
On a very minor point the menu is called England rather than United Kingdom. This is a bit like referring to Belgium as Wallonia!

**Comment by jco-odoo** [comment]
Hi, 

I am currently working on the HMRC MTD VAT Return report for submitting the report electronically.  In the report (enterprise version) right now, there is a box 4B, but I can not find back a box 4B in the documentation of the government and it is calculation is weird, so I would just include it with the rest of box 4.  It is about the reverse charges to e.g. someone without VAT number in the EU.  (if you make an invoice and a credit note, they don't cancel each other in the report, so it seems to be wrong).  

It is weird that in the UK localisation, there is no fiscal position, but in the vat return report, the difference between EC/non-EC seems to be important.  So can we add something like in the screenshot below?
![screenshot from 2018-12-13 17-36-35](https://user-images.githubusercontent.com/3832699/49953221-e6f81880-fefd-11e8-86d7-b65ad619523f.png)

**Comment by StuartJMackintosh** [comment]
From a UK perspective, fiscal positions are only required when trading with non-UK organisations and not usually configured for UK-only trading, yet MTD for VAT is required for all VAT registered companies so not sure that MTD should be implemented within fiscal positions.

When UK trading with EC countries, transaction details are recorded in the EC sales list and not on the VAT return (via box 8). Of course this may all change in coming months...

Where did you find the reference to box 4B and the detail of the weird calculation?

**Comment by jco-odoo** [comment]
Here in the code: https://github.com/odoo/enterprise/blob/12.0/l10n_uk_reports/data/account_financial_html_report_data.xml#L52

We could in theory create a res.country.group for the fiscal positions.  (there might be another way for Europe - uk, need to check)   That way, when you deliver to a uk/non-eu country or another eu country, it will change the tax accordingly instead of needing to change it yourself manually each time.

**Comment by StuartJMackintosh** [comment]
Just a quick note to say that the call for beta testers for the Odoo community MTD project closes 11th Jan -  [details & sign-up here](https://making-tax-digital.software/odoo). Responses have been strong so we are only looking for one or two more participants.

The initial Odoo HMRC authorisation (and 'hello world') module is in alpha-testing and will be released to the beta-testers during January, with the second VAT-specific module in February.

This approach also allows for further HMRC modules to be developed to use the same base HMRC authentication module ie corp tax, automatic EC sales lists etc.

**Comment by jco-odoo** [comment]
@StuartJMackintosh We will be changing the tags anyways in the community version for master, so https://github.com/odoo-dev/odoo/commit/10f5fccb46721583d1f6674898cf57c0c04cd0b7 should make it easier to check the link between taxes and boxes.  We are doing the report the way it is foreseen.  We can not predict the future.

**Comment by StuartJMackintosh** [comment]
@jco-odoo - Yes, hard to predict just now. The tag changes look reasonable at first glance and this may help with the module. @ManishS-OpusVL - please consider the localisation changes in the previous post when testing the MTD.

Lets make sure we are joined up anyway so the OpusVL MTD work and Odoo changes work together. Would you be open to joining the beta-test programme and test with Odoo enterprise?

**Comment by fclementic2c** [comment]
Hello @jco-odoo,

Where do we stand on MTS please.
On on my customer is requesting it urgently for the 7th of May :\

Thanks for your feeback

*camptocamp*

INNOVATIVE SOLUTIONS
BY OPEN SOURCE EXPERTS

*Frédéric Clementi*
Project Manager
Business Solutions

+41 21 619 10 41

www.camptocamp.com

**Comment by jco-odoo** [comment]
@fclementic2c What do you mean by MTS?  MTD is merged.

**Comment by StuartJMackintosh** [comment]
Hello @fclementic2c,

I assume you are referring to MTD.

OpusVL have MTD modules for community & enterprise which were reviewed by HMRC this week and passed the tests. This will be available on the Odoo marketplace over the next week or so and work with V8 to V11 following a beta test with over 20 applicants who have provided valuable feedback and suggestions. 

Features of the OpusVL MTD include:
- automatic VAT reconciliation
- checksumming to ensure data integrity 
- full coverage of HMRC VAT functionality
- direct integration to HMRC - doesn't use a 3rd party
- covers V8 to V11 (V12 will be ported shortly by customer demand)
- works with enterprise and community
- HMRC recognised solution
- beta-tester community developed and tested
- More info on this can be found here: https://making-tax-digital.software/odoo

There are also two other Odoo UK MTD options:
- Odoo S.A. have approved their enterprise integration via a broker (Avalara)
- OdooMTD have a SaaS solution

**Comment by jco-odoo** [comment]
@StuartJMackintosh What you are telling is just wrong.  We did almost the same process and integrated directly. Of course, as we put things in the standard, we focus on usability first.  Available for free on the SaaS.

**Comment by StuartJMackintosh** [comment]
@jco-odoo 

I apologise if the note was not correct -  I was not able to find any other reference to the implementation except for this: https://www.avalara.com/us/en/blog/2016/07/avalara-partners-with-odoo-to-automate-communications-tax-management.html

I couldn't find anything further on the Odoo website regarding MTD and the link from HMRC goes to to Odoo accounting overview page. 

Can you send a link or any further info on the Odoo S.A. implementation as it would be useful for the users to have details of all of the options and a fair view of the differences so that they can choose the best option for them. I will also update my earlier note (assuming that is possible).

During the HMRC review, the assessor made the comment that Odoo was the only product that had multiple suppliers working on an MTD implementation and that they felt that this was very valuable - they explained that having competing MTD solutions was a significant benefit to the users which overall is good for the Odoo product. Lets make sure that we work together and present all of the options.

**Comment by fclementic2c** [comment]
I confirm that I was talking about MTD (not MTS - sorry for the
misspelling. I was writting from my smartphone...)
Thank you for your feedbacks guys.
As my customer is already enterprise we will go for the odoo standard and
see if it fits.




*camptocamp*

INNOVATIVE SOLUTIONS
BY OPEN SOURCE EXPERTS

*Frédéric Clementi*
Project Manager
Business Solutions

+41 21 619 10 41

www.camptocamp.com

**Comment by ManishS-OpusVL** [comment]
Dear all,

This is to formally announce that our Making Tax Digital (MTD) for VAT solution is now been completed and recognized by HMRC.

For more details of our 'MTD Compliant - Odoo' solution please checkout this page https://making-tax-digital.software/odoo

Thank you !

Manish

**Comment by ManishS-OpusVL** [comment]
Dear Community Members,

Would like to update here that we have now released our HMRC recognized MTD modules for versions 8, 10, 11 & 12 on Odoo App store.

Here are the links:
https://apps.odoo.com/apps/modules/12.0/account_mtd_vat/

https://apps.odoo.com/apps/modules/11.0/account_mtd_vat/

https://apps.odoo.com/apps/modules/10.0/account_mtd_vat/

https://apps.odoo.com/apps/modules/8.0/account_mtd_vat/

Thank you !

**Comment by pedrobaeza** [comment]
Should this issue keep opened?

**Comment by jco-odoo** [comment]
@pedrobaeza Those localization issues can stay open to have a general discussion about the localization about what needs to be done next.

**Comment by pedrobaeza** [comment]
OK

**Comment by C3POdoo** [comment]
Dear @lap-odoo,

Thank you for your report but we are closing it due to inactivity.
We apology if we could not look at your request in time.
If your report still makes sense, don't hesitate to reopen a new one. We will try to check it as soon as possible.

_This is an automated message._
